sites/partners/src/listings/PaperListingForm/UnitsSummaryForm.tsx

```diff
@@ -0,0 +1,441 @@
+import React, { useEffect, useState } from "react"
+import {
+  t,
+  GridSection,
+  ViewItem,
+  GridCell,
+  Field,
+  Select,
+  AppearanceStyleType,
+  AppearanceBorderType,
+  FieldGroup,
+  Button,
+  Form,
+  numberOptions,
+} from "@bloom-housing/ui-components"
+import { useForm } from "react-hook-form"
+import { TempUnitsSummary } from "."
+import { UnitAccessibilityPriorityType, UnitType } from "@bloom-housing/backend-core/types"
+import { useUnitPriorityList, useUnitTypeList } from "../../../lib/hooks"
+import { arrayToFormOptions, getRentTypeFromUnitsSummary } from "../../../lib/helpers"
+
+type UnitsSummaryFormProps = {
+  onSubmit: (unit: TempUnitsSummary) => void
+  onClose: () => void
+  summaries: TempUnitsSummary[]
+  currentTempId: number
+}
+
+const UnitsSummaryForm = ({
+  onSubmit,
+  onClose,
+  summaries,
+  currentTempId,
+}: UnitsSummaryFormProps) => {
+  const [current, setCurrent] = useState<TempUnitsSummary>(null)
+  const [tempId, setTempId] = useState<number | null>(null)
+  const [options, setOptions] = useState({
+    unitPriorities: [],
+    unitTypes: [],
+  })
+
+  /**
+   * fetch form options
+   */
+  const { data: unitPriorities = [] } = useUnitPriorityList()
+  const { data: unitTypes = [] } = useUnitTypeList()
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, watch, errors, trigger, getValues, reset } = useForm({
+    defaultValues: {
+      unitType: current?.unitType,
+      floorMin: current?.floorMin,
+      floorMax: current?.floorMax,
+      sqFeetMin: current?.sqFeetMin,
+      sqFeetMax: current?.sqFeetMax,
+      minOccupancy: current?.minOccupancy,
+      maxOccupancy: current?.maxOccupancy,
+      amiPercentage: current?.amiPercentage,
+      minimumIncomeMin: current?.minimumIncomeMin,
+      monthlyRentMin: current?.monthlyRentMin,
+      monthlyRentMax: current?.monthlyRentMax,
+      monthlyRentAsPercentOfIncome: current?.monthlyRentAsPercentOfIncome,
+      priorityType: current?.priorityType,
+      rentType: getRentTypeFromUnitsSummary(current),
+    },
+  })
+
+  useEffect(() => {
+    setTempId(currentTempId)
+  }, [currentTempId])
+
+  useEffect(() => {
+    const summary = summaries.filter((summary) => summary.tempId === tempId)[0]
+    setCurrent(summary)
+    reset({
+      ...summary,
+      rentType: getRentTypeFromUnitsSummary(summary),
+    })
+  }, [summaries, setCurrent, tempId, reset, options])
+
+  const rentType = watch("rentType")
+
+  async function onFormSubmit(action?: string) {
+    const validation = await trigger()
+
+    if (!validation) return
```

Comment:
This differs from the other validation I'm familiar with - can you put a comment here explaining a bit?

Reply:
Done